src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -590,4 +586,49 @@ namespace winrt::TerminalApp::implementation
             VisualStateManager::GoToState(item, L"Normal", true);
         }
     }
+
+    TabCloseButtonVisibility TabBase::CloseButtonVisibility()
+    {
+        return _closeButtonVisibility;
+    }
+
+    void TabBase::CloseButtonVisibility(TabCloseButtonVisibility visibility)
+    {
+        _closeButtonVisibility = visibility;
+        _updateIsClosable();

```

Comment:
Does changing the readonly state call this? I didn't see it in the diffs, but this function is new.

Reply:
Oh, I don't know how I like that. `ReadOnly` doesn't directly impact whether it's closeable unless it's a terminal tab and `ReadOnly` was set by recalculation.

Shouldn't setting ReadOnly automatically trigger this?

Reply:
`_RecalculateAndApplyReadOnly` gets called any time the active pane changes, or any time a control's `ReadOnly` state changes. So setting `ReadOnly` on the pane will set it on the control, who will tell the tab, who will then call `_RecalculateAndApplyReadOnly`